### ISSUE TITLE
readyState constant changes on XMLHttpRequest break ajax calls

### ISSUE THREAD

**Issue by dalethedeveloper** [SOASTA/boomerang#109]
When using the mPulse system with BOOMR 0.9.1434562897 the base XMLHttpRequest is modified to where the state constants are no longer present.  This prevents future references to the parent object for comparisons of readyState.

https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/readyState

Reference to #45 which apparently was not merged as stated when closed Jan 28, 2015

Standard XMLHttpRequest
```
> XMLHttpRequest
 function XMLHttpRequest() { [native code] }
> XMLHttpRequest.DONE
4
>var x = new XMLHttpRequest()
> x.DONE
4
```

With mPulse Boomerang modified XMLHttpRequest
```
> XMLHttpRequest
(){var a,c,d,e={timing:{},initiator:"xhr"};a=new BOOMR.orig_XMLHttpRequest;…
> XMLHttpRequest.DONE
undefined
>var x = new XMLHttpRequest()
> x.DONE
4
```

This causes code that performs the following type of evaluations to fail silently, further impeding efforts to troubleshoot.
```
if(xmlhttp.readyState == XMLHttpRequest.DONE){
```

**Comment by dalethedeveloper** [comment]
Workaround for affected code is to use numeric constant:
```
if(xmlhttp.readyState == (XMLHttpRequest.DONE || 4))
```

**Comment by bluesmoon** [comment]
0.9.1434562897 is an obsolete version.  Can you request your app be upgraded to a more recent build?

**Comment by nicjansma** [comment]
@dalethedeveloper This is fixed in Boomerang version 1.331 or higher.  Please let us know if it's not resolved with an app upgrade.
